Nos/Views/RepliesView.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  ThreadView.swift
+//  Nos
+//
+//  Created by Matthew Lorentz on 2/14/23.
+//
+
+import SwiftUI
+import SwiftUINavigation
+
+struct RepliesView: View {
+    @Environment(\.managedObjectContext) private var viewContext
+    
+    @EnvironmentObject private var relayService: RelayService
+    @EnvironmentObject var router: Router
+
+    @State private var reply = ""
+    
+    @State private var alert: AlertState<Never>?
+    
+    var repliesRequest: FetchRequest<Event>
+    var replies: FetchedResults<Event> { repliesRequest.wrappedValue }
+    
+    var directReplies: [Event] {
+        replies.filter { ($0.eventReferences?.lastObject as? EventReference)?.eventId == note.identifier }
+    }
+    
+    init(note: Event) {
+        self.note = note
+        
+        if let rootReference = (note.eventReferences?.array as? [EventReference])?
+            .first(where: { $0.marker == "root" }),
+            let rootId = rootReference.eventId {
+            self.repliesRequest = FetchRequest(fetchRequest: Event.allReplies(toEventWith: rootId))
+        } else {
+            self.repliesRequest = FetchRequest(fetchRequest: Event.allReplies(to: note))
+        }
+    }
+    
+    private var keyPair: KeyPair? {
+        KeyPair.loadFromKeychain()
+    }
+    
+    var note: Event
+    
+    var body: some View {
+        ZStack {
+            ScrollView(.vertical) {
+                LazyVStack {
+                    NoteButton(note: note)
+                        .padding(.horizontal)
+                    ForEach(directReplies.reversed()) { event in
+                        ThreadView(root: event, allReplies: replies.reversed())
+                    }
+                }
+            }
+            .padding(.top, 1)
+            .navigationBarBackButtonHidden(true)
+            .onAppear {
+                router.navigationTitle = Localized.threadView.rawValue
+            }
+            VStack {
```

Comment:
I noticed an issue with this VStack, not sure if it's new in your PR so if it isn't feel free to open a bug ticket. But in a long thread I can't scroll the last reply full into view because the scroll view extends under the reply text field. I think this could be fixed by changing the enclosing ZStack to a VStack maybe?

Reply:
Yeah I think this might have been an issue before this ticket (not sure), but I'll try and fix it